tla-bmcmt/src/main/scala/at/forsyte/apalache/tla/bmcmt/config/CheckerExceptionAdapter.scala

```diff
@@ -28,7 +28,14 @@ class CheckerExceptionAdapter @Inject() (sourceStore: SourceStore, changeListene
     extends ExceptionAdapter with LazyLogging {
   private lazy val ISSUES_LINK: String = "[https://github.com/informalsystems/apalache/issues]"
 
-  override def toMessage: PartialFunction[Exception, ErrorMessage] = {
+  override def toMessage: PartialFunction[Throwable, ErrorMessage] = {
+    // system issues
+    case err: OutOfMemoryError =>
+      logger.error(s"Ran out of heap memory (max JVM memory: ${Runtime.getRuntime.maxMemory})")
+      logger.error(s"To increase available heap memory, see the manual:")
+      logger.error("  [https://apalache.informal.systems/docs/apalache/running.html#supplying-jvm-arguments]")
+      NormalErrorMessage(s"Ran out of heap memory: ${err.getMessage}")
```

Comment:
The body of this `case` seems to repeat 3 times in this PR, can you make an aux method call for this?

Reply:
Actually, let me just provide that in the parent trait...

Reply:
In the context of [this](https://apalache.informal.systems/docs/adr/007adr-restructuring.html), it feels like an `apa-base/glue` method, but practically, probably `tla-types` or even `lir`

Reply:
Yeah, in the trait is probably easiest, didn't even notice they all inherit the same

Reply:
Addressed in [493260d](https://github.com/informalsystems/apalache/pull/1711/commits/493260d548d9e422f4360219c436f4006222535f)